test/Examples.hs

```diff
@@ -38,16 +39,17 @@ main = withOptions opts $ \ config logger statter -> hspec . parallel $ do
   runIO setupExampleRepos
 
   for_ languages $ \ lang@LanguageExample{..} -> do
-    let tsLang = "tree-sitter-" <> languageName
-        tsDir = languagesDir </> tsLang </> "vendor" </> tsLang
+    let tsLang = Path.relDir ("tree-sitter-" <> languageName)
+        tsDir = languagesDir </> tsLang </> Path.relDir "vendor" </> tsLang
     parallel . describe languageName $ parseExamples args lang tsDir
 
   where
     parseExamples session LanguageExample{..} tsDir = do
       knownFailures <- runIO $ knownFailuresForPath tsDir languageKnownFailuresTxt
-      files <- runIO $ globDir1 (compile ("**/*" <> languageExtension)) (tsDir </> languageExampleDir)
-      for_ files $ \file -> it file $ do
-        res <- runTask session (parseFilePath file)
+      files <- runIO $ globDir1 (compile ("**/*" <> languageExtension)) (Path.toString (tsDir </> languageExampleDir))
+      let paths = Path.relFile <$> files
+      for_ paths $ \file -> it (Path.toString file) $ do
+        res <- runTask session (parseFilePath (Path.toString file))
```

Comment:
This particular instance feels a little counterproductive, but I suppose that’ll be improved when `parseFilePath` is amended to take a typed path.

Reply:
Oh, lemme do that now.